### DISCUSSION TITLE
Can't get lastMessage from other users

### DISCUSSION THREAD

**Discussion by TheMathGeek314** [discordjs/discord.js#5428] [Q&A] [answered]
I'm trying to retrieve the latest message of a particular user by mentioning them, but it only works if I mention myself or if they mention themself. According to the console, the user's lastMessageId is null even though they sent a message recently. Is there some permission in discord I'm missing? Is this a bug?

**Comment by DaStormer** [comment]
I believe `lastMessageID` is dependent on cache, meaning it'd only work if the message in question is cached.

**Reply by TheMathGeek314** [reply]
Is there a way to cache older messages or somehow get around that?

**Reply by DaStormer** [reply]
Have you set any [ClientOptions](https://discord.js.org/#/docs/main/stable/typedef/ClientOptions) that control the message cache? By default it should cache the last 200 messages sent per channel, unless you've altered that.

**Reply by TheMathGeek314** [reply]
I haven't messed with anything, so it should be defaults. However, I can't seem to save the "bot permissions" checklist, which does include message history. I check what I want, refresh, and it's empty

**Reply by DaStormer** [reply]
Are you referring to the one in the Discord Developer Portal? If so, that's just to calculate a permissions integer. 

The actual bot permissions in your server are configured through the regular Discord role/channel permissions in the server.

**Comment by tipakA** [comment, marked as answer]
`lastMessageID` is dependant on cache. Cache is empty when you start the bot, and API does not send you every message from every channel when you log your bot in. Library caches messages that were received since the bot started up, but to get anything beyond that, you have to fetch messages.

You cannot fetch messages from specific users (and search is not available for bot accounts). You also cannot fetch messages from multiple channels at once, so to get *actually* most recent message from a user, you would have to fetch unspecified amount of messages from unspecified amount of channels, in worst case ending up with fetching all messages in a server and getting no results.

There isn't really a reliable way to get last message a user has sent.
